src/script/lua_api/l_object.cpp

```diff
@@ -526,11 +527,19 @@ int ObjectRef::l_set_bone_position(lua_State *L)
 	if (sao == nullptr)
 		return 0;
 
-	std::string bone = readParam<std::string>(L, 2, "");
-	v3f position = readParam<v3f>(L, 3, v3f(0, 0, 0));
-	v3f rotation = readParam<v3f>(L, 4, v3f(0, 0, 0));
-
-	sao->setBonePosition(bone, position, rotation);
+	std::string bone;
+	if (!lua_isnil(L, 2))
+		bone = readParam<std::string>(L, 2);
+	BoneOverride *override = new BoneOverride();
+	if (!lua_isnil(L, 3))
+		override->position.vector = check_v3f(L, 3);
+	if (!lua_isnil(L, 4))
+		override->rotation.next = core::quaternion(check_v3f(L, 4) * core::DEGTORAD);
+	override->position.absolute = true;
+	override->rotation.absolute = true;
+	override->position.interpolation_duration = 0.0f;
+	override->position.interpolation_duration = 0.0f;
```

Comment:
These could be specified as defaults in the `BoneOverride` struct directly.

Reply:
They should be. This code should be unneeded.

Reply:
After giving this a second thought, I have come to the following conclusions:

1. Setting `absolute` to `true` is reasonable as `true` would be an unreasonable default for the struct and this is just for legacy support;
2. The `interpolation_duration` already is zero by default. These lines can be safely removed.